src/db_adapters/receipts.rs

```diff
@@ -17,10 +17,105 @@ pub(crate) async fn store_receipts(
+            loop {
+                if skipping_receipt_ids.is_empty() {
+                    break models;
+                } else {
+                    warn!(
+                        target: crate::INDEXER_FOR_EXPLORER,
+                        "Looking for transaction hashes for receipts again... \n"
+                    );
+                    let tx_hashes_for_receipts = find_tx_hashes_for_receipts(
+                        &pool,
+                        skipping_receipt_ids.iter().map(|r| r.to_string()).collect(),
+                    )
+                    .await;
+                    models.extend(
+                        receipts
+                            .iter()
+                            .filter(|r| skipping_receipt_ids.contains(&r.receipt_id))
+                            .filter_map(|r| {
+                                if let Some(transaction_hash) =
+                                    tx_hashes_for_receipts.get(r.receipt_id.to_string().as_str())
+                                {
+                                    Some(models::Receipt::from_receipt_view(
+                                        r,
+                                        block_hash,
+                                        transaction_hash,
+                                    ))
+                                } else {
+                                    warn!(
+                                        target: crate::INDEXER_FOR_EXPLORER,
+                                        "Skipping Receipt {} as we can't find parent Transaction for it.",
+                                        r.receipt_id.to_string()
+                                    );
+                                    new_skipping_receipt_ids.insert(r.receipt_id);
+                                    None
+                                }
+                            }),
+                    );
+                    skipping_receipt_ids = new_skipping_receipt_ids.clone();
+                    new_skipping_receipt_ids.clear()
+                }
+            }
+        };
+
+        receipt_models.extend_from_slice(&receipt_models_);
+    }
+
+    save_receipts(&pool, receipt_models).await;
+
+    let (action_receipts, data_receipts): (Vec<&near_indexer::near_primitives::views::ReceiptView>, Vec<&near_indexer::near_primitives::views::ReceiptView>) = receipts
+        .into_iter()
+        .filter(|r| !skipping_receipt_ids.contains(&r.receipt_id))
+        .partition(|receipt| matches!(receipt.receipt, near_indexer::near_primitives::views::ReceiptEnumView::Action { .. }));
+
+    let process_receipt_actions_future = store_receipt_actions(&pool, action_receipts);
+
+    let process_receipt_data_future = store_receipt_data(&pool, data_receipts);
+
+    join!(process_receipt_actions_future, process_receipt_data_future);
+}
+
+/// Looks for already created parent transaction hash for given receipts
+async fn find_tx_hashes_for_receipts(
```

Comment:
I suggest this function takes all the hassle about the strict mode, so you collect all the transaction hashes first and then you create all the receipt models in a single go.